packages/test/test-end-to-end-tests/src/test/sharedTreeEndToEndTests.spec.ts

```diff
@@ -0,0 +1,461 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { describeCompat } from "@fluid-private/test-version-utils";
+import type { IContainer } from "@fluidframework/container-definitions/internal";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { ITestObjectProvider } from "@fluidframework/test-utils/internal";
+import {
+	ITree,
+	SchemaFactory,
+	TreeViewConfiguration,
+	type TreeView,
+} from "@fluidframework/tree";
+import {
+	asTreeViewAlpha,
+	SharedTree,
+	type Revertible,
+	type TreeViewAlpha,
+} from "@fluidframework/tree/internal";
+
+describeCompat("SharedTree", "NoCompat", (getTestObjectProvider, apis) => {
+	const { DataObject, DataObjectFactory } = apis.dataRuntime;
+	const { ContainerRuntimeFactoryWithDefaultDataStore } = apis.containerRuntime;
+
+	// An extension of Aqueduct's DataObject that creates a SharedTree during initialization and exposes it.
+	class DataObjectWithTree extends DataObject {
+		private readonly treeKey = "tree";
+
+		private _tree: ITree | undefined;
+		public get tree(): ITree {
+			assert(this._tree !== undefined, "Tree not initialized");
+			return this._tree;
+		}
+
+		protected async initializingFirstTime() {
+			const tree = SharedTree.create(this.runtime);
+			this.root.set(this.treeKey, tree.handle);
+		}
+
+		protected async hasInitialized() {
+			const treeHandle = this.root.get<IFluidHandle<ITree>>(this.treeKey);
+			assert(treeHandle, "Tree handle not found");
+			this._tree = await treeHandle.get();
+		}
+	}
+	// A data object factory that creates DataObjectWithTree instances.
+	const dataObjectFactoryWithTree = new DataObjectFactory(
+		"DataObjectWithTree",
+		DataObjectWithTree,
+		[SharedTree.getFactory()],
+		{},
+	);
+
+	// Runtime factory that can create containers with DataObjectWithTree instances via the above data object factory.
+	const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore({
+		defaultFactory: dataObjectFactoryWithTree,
+		registryEntries: [
+			[dataObjectFactoryWithTree.type, Promise.resolve(dataObjectFactoryWithTree)],
+		],
+		runtimeOptions: { enableRuntimeIdCompressor: "on" }, // Needed to create a shared tree
+	});
+
+	let provider: ITestObjectProvider;
+	let container1: IContainer;
+	let container2: IContainer;
+	let dataObject1: DataObjectWithTree;
+	let tree1: ITree;
+	let tree2: ITree;
+
+	beforeEach("setup", async function () {
+		provider = getTestObjectProvider();
```

Comment:
This is going to pull in an unnecessary map in addition to the directory used above. https://github.com/microsoft/FluidFramework/pull/23949 provides an option to avoid that.

Reply:
I don't think it will because I am using my own runtime factory with a DataObject that should only pull in a SharedDirectory.

Reply:
I had assumed TestObjectProvider would provide a TestFluidObject. Looks like I was incorrect with that assumption.